File: lib/data/sockets/tournament_content_socket.dart
import 'package:flutter/foundation.dart';
import 'package:seating_generator_web/data/base_socket.dart';

class TournamentContentSocket extends BaseSocket {
  TournamentContentSocket({required int tournamentId, required int table})
      : super(
          '${getScheme()}://${getHost()}/api/seatingContent?table=$table&tournamentId=$tournamentId',
        );
}

String getScheme() {
  if (kIsWeb && kReleaseMode) {
    return Uri.base.scheme == 'https' ? 'wss' : 'ws';
  }

  return 'wss';
}

String getHost() {
  if (kIsWeb && kReleaseMode) {
    return '${Uri.base.host}:${Uri.base.port}';
  }

  return 'mafbase.ru';
}

File: mafbase_stream/ios/Classes/Overlay/Plashki/TournamentContentSocket.swift
import Foundation

/// Подписка на seatingContent через WebSocket. Кодовый аналог Kotlin-класса
/// `TournamentContentSocket.kt` (см. android/.../data/sockets/) и Dart-класса
/// `TournamentContentSocket` (см. lib/data/sockets/tournament_content_socket.dart).
/// Вся механика соединения — в базовом [SeatingContentSocket].
final class TournamentContentSocket: SeatingContentSocket {

    init(tournamentId: Int, table: Int) {
        var components = URLComponents(string: "wss://mafbase.ru/api/seatingContent")!
        components.queryItems = [
            URLQueryItem(name: "table", value: String(table)),
            URLQueryItem(name: "tournamentId", value: String(tournamentId)),
        ]
        super.init(url: components.url!, logTag: "TournamentSocket")
    }
}
